http/v1.md

```diff
@@ -0,0 +1,493 @@
+
+```
+{
+    "code": 101,
+    "reason": "Validation failed",
+    "validationErrors": [
+        {
+            "field": "maker",
+            "code": 1002,
+            "reason": "Invalid address"
+        }
+    ]
+}
+```
+
+General error codes:
+
+```
+100 - Validation Failed
+101 - Malformed JSON
+102 - Order submission disabled
+103 - Throttled
+```
+
+Validation error codes:
+
+```
+1000 - Required field
+1001 - Incorrect format
+1002 - Invalid address
+1003 - Address not supported
+1004 - Value out of range
+1005 - Invalid signature or hash
+1006 - Unsupported option
+```
+
+
+### Misc.
+
+*   All requests and responses should be of **application/json** content type
+*   All token amounts are sent in amounts of the smallest level of precision (base units). (e.g if a token has 18 decimal places, selling 1 token would show up as selling `'1000000000000000000'` units by this API).
+*   All addresses are sent as lower-case (non-checksummed) Ethereum addresses with the `0x` prefix.
+*   All URL query parameters are to be written in `snake_case` and all queries and responses specified in JSON should use `lowerCamelCase`.
+
+## REST API
+
+### GET /v1/asset_pairs
+
+Retrieves a list of available asset pairs and the information required to trade them. This endpoint should be [paginated](#pagination).
+
+#### Parameters
+
+*   assetDataA=&assetDataB [string]: these are assetData fields. Returns asset pairs that contain assetDataA and assetDataB (in any order). Setting only assetDataA or assetDataB returns pairs filtered by that asset only (optional)
+
+#### Response
+
+[See response schema](https://github.com/0xProject/0x-monorepo/blob/development/packages/json-schemas/schemas/relayer_api_asset_pairs_response_schema.ts#L1)
+
+```
+{
+    "total": 43,
+    "page": 1,
+    "perPage": 100,
+    "records": [
+        {
+            "assetDataA": {
+                "minAmount": "0",
+                "maxAmount": "10000000000000000000",
+                "precision": 5,
+                "assetData": "0xf47261b04c32345ced77393b3530b1eed0f346429d"
+            },
+            "assetDataB": {
+                "minAmount": "0",
+                "maxAmount": "50000000000000000000",
+                "precision": 5
+                "assetData": "0x0257179264389b814a946f3e92105513705ca6b990"
+            }
+        },
+        ...
+    ]
+}
+```
+
+*   `minAmount` - the minimum trade amount the relayer will accept
+*   `maxAmount` - the maximum trade amount the relayer will accept
+*   `precision` - the desired price precision a relayer would like to support within their orderbook
+*   `assetData` - the assetData representing that token
+
+### GET /v1/orders
+
+Retrieves a list of orders given query parameters. This endpoint should be [paginated](#pagination). For querying an entire orderbook snapshot, the [orderbook endpoint](#get-v1orderbook) is recommended.
+
+#### Parameters
+
+Filter parameters:
```

Comment:
At the risk of complicating things, all of these filters could be combined into a query with the following params:

- `makerAssetData` or `takerAssetData`
- a byte range within the `assetData` (e.g 0-4 for proxy id, 16-36 for address). This would probably be broken down into `startIndex` and `endIndex`
- the target bytes

This is a general way to filter by proxyId, asset address, assetId of a NFT, or any other params in new token standards.